qcodes/instrument_drivers/AlazarTech/ATS.py

```diff
@@ -962,3 +947,44 @@ def buffer_done_callback(self, buffers_completed):
                 to local memory at the time of this callback.
         """
         pass
+
+
+class AcquisitionController(Instrument, AcquisitionInterface):
+    """
+    This class represents all choices that the end-user has to make regarding
+    the data-acquisition. this class should be subclassed to program these
+    choices.
+
+    The basic structure of an acquisition is:
+
+        - Call to AlazarTech_ATS.acquire internal configuration
+        - Call to acquisitioncontroller.pre_start_capture
+        - Call to the start capture of the Alazar board
+        - Call to acquisitioncontroller.pre_acquire
+        - Loop over all buffers that need to be acquired
+          dump each buffer to acquisitioncontroller.handle_buffer
+          (only if buffers need to be recycled to finish the acquisiton)
+        - Dump remaining buffers to acquisitioncontroller.handle_buffer
+          alazar internals
+        - Return acquisitioncontroller.post_acquire
+
+    Attributes:
+        _alazar: a reference to the alazar instrument driver
+    """
+
+    def __init__(self, name, alazar_name, **kwargs):
+        """
+        Args:
+            alazar_name: The name of the alazar instrument on the server
+        """
+        super().__init__(name, **kwargs)
+        self._alazar = self.find_instrument(alazar_name,
+                                            instrument_class=AlazarTech_ATS)
+
+    def _get_alazar(self):
```

Comment:
We should really deprecate this. But that is for another time

Reply:
fully agree